config/routes.rb

```diff
@@ -2,18 +2,23 @@
   devise_for :users
   resources :users, only: [:index, :show] do
     resources :posts, only: [:index, :show, :new, :create, :destroy] do
+      member do
+        post 'likes' => 'posts#likes', as: :likes
+      end
       resources :comments, only: [:create, :new, :destroy, :index]
-      resources :likes, only: [:create]
     end
-  end 
+  end
 
-  root 'users#index'
-end
-
-    # resources :posts, only: [:index, :show, :new, :create, :destroy] do
-    #   member do
-    #     post 'likes' => 'posts#likes', as: :likes
-    #   end
-    #   resources :comments, only: [:create, :new, :destroy, :index]
-    # end
-  
\ No newline at end of file
+  namespace :api do
+    namespace :v1 do
+      resources :users, only: [] do
+        resources :posts, only: [:index]
+      end
+      resources :posts, only: [] do
+        resources :comments, only: [:index, :create]
+      end
+    end
+  end
+  
```

Comment:
- [ ] After the issue highlighted about the `comment_params` method I am still unable to create a comment. If you look at the comment `create` action you are checking for a `user_id` parameter from the URL but you did not build route properly to take that.

  Your current create comment route is this:
  ```
  http://localhost:3000/api/v1/posts/:post_id/comments
  ```
  But it should be this:
  ```
  http://localhost:3000/api/v1/users/:user_id/posts/:post_id/comments
  ```
  Kindly refactor this API routes to this:
  ```
  namespace :api do
    namespace :v1 do
      resources :users, only: [] do
        resources :posts, only: [:index] do
          resources :comments, only: [:index, :create]
        end
      end
    end
  end
  ```

  ![idosq](https://user-images.githubusercontent.com/36542986/231748118-4878b5b4-c394-4e1d-baaf-3803b88c0b99.jpg)